assets/js/googlesitekit/datastore/user/key-metrics.js

```diff
@@ -172,22 +173,21 @@ const baseActions = {
 
 const baseControls = {};
 
-const baseReducer = ( state, { type, payload } ) => {
+const baseReducer = createReducer( ( state, action ) => {
+	const { type, payload } = action;
 	switch ( type ) {
 		case SET_KEY_METRICS_SETTING: {
-			return {
-				...state,
-				keyMetricsSettings: {
-					...state.keyMetricsSettings,
-					[ payload.settingID ]: payload.value,
-				},
-			};
+			if ( ! state.keyMetricsSettings ) {
+				state.keyMetricsSettings = {};
+			}
+			state.keyMetricsSettings[ payload.settingID ] = payload.value;
+			break;
 		}
 		default: {
 			return state;
```

Comment:
We don't need to return from an Immer reducer:

```suggestion
			break;
```